### PR TITLE
Feature/dr 2873 short circuit rights lookup

### PR DESCRIPTION
**Pull request by emu47**
## Ticket:

Short circuit rights lookup for homepage images. [DR-2873](https://jira.nypl.org/browse/DR-2873).

## Things Done

- Added a bypass for image ids used in the homepage to pull rights from pre-cooked rights information stored in a hash.
- Added unit tests

## How to Test

- Locally, run unit tests in jruby: `jruby test/delegates_test.rb`
- Run container locally and test image responses for image_ids from lanes data found here: https://github.com/NYPL/digital-collections/blob/main/src/data/lanes.ts
- Go to the logs in AWS console and search for any sign of the homepage image_ids. You should see no activity in repo api for those image_ids. 

## Concerns

- This method should eventually be removed once Repo API responses are improved. This is not meant to be a longterm solution.

### CODE REVIEW

File: delegates.rb [hunk 1]
```diff
@@ -257,13 +262,45 @@ def fetch(path, ip)
...
     response = http.request(request)
-
     return response.body
   end
 
   def api_url(path)
     "https://api.repo.nypl.org/api/v2/#{path}"
   end
+  
+  def homepage_image_rights_hash
```
Comment (chrisb87) [inline delegates.rb:272]: Making this a frozen constant will make it a bit more performant.

**Review by chrisb87 — CHANGES_REQUESTED** [review]
Since this new code is run for every single request, I think we should do the small performance tweak I suggested.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+### Changed
+- Created a bypass for images features on the DC Facelift homepage to not call Repo API. (DR-2873)
```
Comment (croyfish) [inline CHANGELOG.md:10]: Nitpick, but I would call these image IDs out as swimlanes images in the changelog message so we don't get confused later with the featured images

File: test/delegates_test.rb [hunk 3]
```diff
@@ -3,55 +3,75 @@
 require_relative '../delegates'
```
Comment (croyfish) [inline test/delegates_test.rb:3]: 👍 nice improvements to this file

**Comment by croyfish** [comment]
I added 2 images to my local mock images repo: one in the hash and one not, and then configured a phony repo api URL in delegates. The image in the swim lanes hash renders fine, and the other image returns 403. I think the code changes are working as intended.

File: delegates.rb (hunk 1 above)
Comment (emu47) [reply delegates.rb:272]: Totally great call.

Comment (chrisb87) [reply delegates.rb]: updated

File: CHANGELOG.md (hunk 2 above)
Comment (chrisb87) [reply CHANGELOG.md]: changelog updated 👍

**Review by chrisb87 — APPROVED** [review]

File: test/delegates_test.rb [hunk 4]
```diff
@@ -2,56 +2,78 @@
 require 'minitest/autorun'
 require_relative '../delegates'
 
+# Mocking HOMEPAGE_IMAGE_RIGHTS_HASH constant to return rights for a specific image_id
+class CustomDelegate
+  HOMEPAGE_IMAGE_RIGHTS_HASH = {
```
Comment (chrisb87) [inline test/delegates_test.rb:7]: Just calling out here that this "mocking" method is risky.

If this file gets required, the actual rights hash in the file will be overridden with these mocks. So we must be sure that the production code never actually requires this test file. It doesn't.

**Review by croyfish — APPROVED** [review]
The images still work after the most recent commit.

**Pull request merged**